noir/bootstrap.sh

```diff
@@ -1,9 +1,18 @@
 #!/usr/bin/env bash
 source $(git rev-parse --show-toplevel)/ci3/source_bootstrap
 
+set -eou pipefail
+
 cmd=${1:-}
 [ -n "$cmd" ] && shift
 
+# Update the noir-repo before we hash its content, unless the command is exempt.
+no_update=(clean make-patch bump-noir-repo-ref)
+if [[ -z "$cmd" || ! ${no_update[*]} =~ "$cmd" ]]; then
```

Comment:
Why not just hash the ref and the patch file?

Reply:
I didn't want to fundamentally change what the hashes work with: have we compiled and cached this code or not, including pending changes and local commits that may not be in the hash file.